### ISSUE TITLE
Fix EJS tag just before `<table>` element causes parse error

### ISSUE THREAD

**Issue by SaekiTominaga** [markuplint/markuplint#607]
- Markuplint version: `3.0.0-rc.1`
- Parser lang: EJS (`@markuplint/ejs-parser@3.0.0-rc.1`)
- Node.js version: `18.12.0`
- OS: Windows

## Describe the bug

EJS tag just before `<table>` element causes parse error.
Except for `<table>`, no errors occurred. However, I did not check all HTML elements.
Note, no problem with version 2 (2.11.3).

## Code Example

```html
<% if (foo) { %> <!-- ← error: The #comment is invalid element (N:1): Broke mapping nodes. (parse-error) -->
<table></table>
<% } %>
```
or
```html
<% if (foo) { %>
<p></p>
<% } %> <!-- ← error: The #comment is invalid element (N:1): Broke mapping nodes. (parse-error) -->
<table></table>
```

**Comment by YusukeHirao** [comment]
@SaekiTominaga 

I didn't reproduce that.
There's a possibility that the cause is the context or properties that are line breaks and more.
Could you create and share your repository that includes the problem code?

**Comment by SaekiTominaga** [comment]
@YusukeHirao 
I made a branch to reproduce in the following repository.
https://github.com/SaekiTominaga/markuplint-config/tree/markuplint-v3-ejs-parse-error-table

Branch: `markuplint-v3-ejs-parse-error-table`

When I run `npm run test-parser`, parse-error occurs only for `ejs-if-table.ejs`.

![Screenshot of the result of running the test in the terminal. passed: ejs-if-p.ejs, error: The #comment is invalid element (9:3): Broke mapping nodes. (parse-error): ejs-if-table.ejs, passed: ejs-tags.ejs](https://user-images.githubusercontent.com/4138486/206903733-66b03cfb-6df2-4fc4-93f0-88fcfbdca9f8.png)

**Comment by YusukeHirao** [comment]
I found the cause.
It's the bug caused by combinate the sibling selector and the parser.
I'll fix it.
